app/src/main/res/layout/activity_main.xml

```diff
@@ -13,5 +13,95 @@
     <TextView
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
+        android:layout_alignParentTop="true"
         android:text="Hello Chemist!" />
+
+    <LinearLayout android:layout_height="wrap_content"
+                  android:layout_width="fill_parent"
+                  android:layout_alignParentBottom="true"
+                  android:layout_gravity="center_horizontal"
+                  android:gravity="center_horizontal"
+                  android:weightSum="5" >
+        <Button
+                android:id="@+id/unit_analysis"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:gravity="center"
+                android:paddingBottom="2dp"
+                android:paddingLeft="2dp"
+                android:paddingRight="2dp"
+                android:paddingTop="2dp"
+                android:layout_weight="1"
```

Comment:
From what I can tell, the weights have to be set explicitly to fit all 5 buttons within the space provided.

As for color/text size, I don't believe those were defaults, but I sort of just chose arbitrary values. The sizing was for aesthetics on the emulator, but I believe that's a device-specific issue that will need to be addressed.

Reply:
Interesting. weight might have something to do with how overflow is handled.

I'm picturing buttons stacked vertically on top of each other, which alleviates/eliminates the crowding issue. (Although it could still become an issue later.)

Reply:
Yeah, the weights aren't relevant after converting the layout to vertically stack the buttons. I've removed the `weightSum` and individual `weight` attributes entirely.

Reply:
I've added a NotImplemented activity and linked each button from the main activity to it. The `onClickListener` for each button seemed to be a bit repetitive, so there may be a better approach (or maybe just necessary boilerplate).

I included a home button on the `NotImplemented` page to allow us to return to the home page as well.

Reply:
I added a note that might cut down on redundancy in the `NotImplemented.java` file.

Interestingly, we don't need the "Home" button, since Android phones have a built-in "back" button. Each new activity stacks on top of the previous one, so you can get back to Home by just hitting the back button. (I know this differs from typical Apple behavior, so your instincts aren't really surprising.)

What this button technically does is create a *new* home page, and stack it on top of the existing home and not-implemented pages.

I'd recommend removing the button to simplify the activity layout, prevent weirdly-stacked activities, and to allow us to link to the NotImplemented activity from places that might not be the home screen.

Reply:
I didn't realize all handsets supported a physical back button. In any case, I have removed the home button from the NotImplementedActivity.